easybuild/easyconfigs/v/VTK/VTK-9.1.0-foss-2021b.eb

```diff
@@ -0,0 +1,86 @@
+##
+# Authors::
+# * Fotis Georgatos <fotis@cern.ch>
+# * Robert Mijakovic <robert.mijakovic@lxp.lu>
+##
+
+easyblock = 'CMakeMake'
+
+name = 'VTK'
+version = '9.1.0'
+
+homepage = 'https://www.vtk.org'
+description = """The Visualization Toolkit (VTK) is an open-source, freely available software system for
+ 3D computer graphics, image processing and visualization. VTK consists of a C++ class library and several
+ interpreted interface layers including Tcl/Tk, Java, and Python. VTK supports a wide variety of visualization
+ algorithms including: scalar, vector, tensor, texture, and volumetric methods; and advanced modeling techniques
+ such as: implicit modeling, polygon reduction, mesh smoothing, cutting, contouring, and Delaunay triangulation."""
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+toolchainopts = {'usempi': True}
+
+source_urls = ['https://www.vtk.org/files/release/%(version_major_minor)s']
+sources = [
+    SOURCE_TAR_GZ,
+    '%(name)sData-%(version)s.tar.gz',
+]
+patches = [('vtk-version.egg-info', '.')]
+checksums = [
+    '8fed42f4f8f1eb8083107b68eaa9ad71da07110161a3116ad807f43e5ca5ce96',  # VTK-9.1.0.tar.gz
+    'b9442cf1c30e1e44502e6dc36d3c6c2dc3d3f7d03306ee1d10737e9abadaa85d',  # VTKData-9.1.0.tar.gz
+    '787b82415ae7a4a1f815b4db0e25f7abc809a05fc85d7d219627f3a7e5d3867b',  # vtk-version.egg-info
+]
+
+builddependencies = [('CMake', '3.21.1')]
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('XZ', '5.2.5'),
+    ('libGLU', '9.0.2'),
+    ('X11', '20210802'),
+]
+
+separate_build_dir = True
+
+# OpenGL
+configopts = "-DOPENGL_glu_LIBRARY=$EBROOTLIBGLU/lib/libGLU.%s " % SHLIB_EXT
+configopts += "-DOPENGL_gl_LIBRARY=$EBROOTMESA/lib/libGL.%s " % SHLIB_EXT
+configopts += "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include "
```

Comment:
Need to pick up the added deps with
```
configopts += "-DVTK_USE_SYSTEM_EXPAT:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_FREETYPE:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_GL2PS:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_HDF5:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_JSONCPP:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_LIBXML2:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_LZ4:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_MPI4PY:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_NETCDF:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_NETCDFCPP:BOOL=OFF "
configopts += "-DVTK_USE_SYSTEM_PNG:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_TIFF:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_JPEG:BOOL=ON "
configopts += "-DVTK_USE_SYSTEM_ZLIB:BOOL=ON -DZLIB_INCLUDE_DIR:PATH=$EBROOTZLIB "
```

Reply:
And for Qt5 may need
```
configopts += "-DVTK_QT_VERSION=5 "
```

Reply:
For the future benefit of Visit should also have
```
configopts += "-DVTK_OPENGL_HAS_OSMESA:BOOL=ON "
configopts += "-DOSMESA_LIBRARY:FILEPATH=$EBROOTMESA/lib/libOSMesa.%s " % SHLIB_EXT
configopts += "-DOSMESA_INCLUDE_DIR:PATH=$EBROOTMESA/include "
```

and 
```
configopts += "-DVTK_LEGACY_REMOVE:BOOL=true -DVTK_REPORT_OPENGL_ERRORS:BOOL=true "
configopts += "-DVTK_ALL_NEW_OBJECT_FACTORY:BOOL=true "
configopts += "-DModule_vtkGUISupportQtOpenGL:BOOL=true "
configopts += "-DModule_vtklibxml2:BOOL=true "
```

Reply:
However, the current version of Visit (3.2.2) can't use VTK > 8.1 at the moment. (Still working on that)
So the Visit related parts aren't necessary at the moment.